File: 240127/행복한 수열의 개수/number-of-happy-sequence.py
n, m = map(int, input().split())

grid = [list(map(int, input().split())) for _ in range(n)]

def is_happy():
    same_cnt, max_cnt = 1, 1
    for i in range(1, n):
        if(seq[i - 1] == seq[i]):
            same_cnt += 1
        else:
            same_cnt = 1
        
        max_cnt = max(max_cnt, same_cnt)
    return max_cnt >= m

cnt = 0

for i in range(n):
    seq = grid[i][:]
    
    if is_happy():
        cnt += 1

for j in range(n):
    for i in range(n):
        seq[i] = grid[i][j]
    if is_happy():
        cnt += 1

print(cnt)

File: 240127/행복한 수열의 개수/number-of-happy-sequence.java
import java.util.Scanner;

public class Main {
    public static final int MAX_N = 100;
    public static int n, m;
    public static int[][] grid = new int[MAX_N][MAX_N];

    public static int[] seq = new int[MAX_N];
    
    public static boolean isHappy() {
        int sameCnt = 1, maxCnt = 1;
        for(int i = 1; i<n; i++){
            if(seq[i-1] == seq[i])
                sameCnt++;
            else
                sameCnt = 1;
            
            maxCnt = Math.max(maxCnt, sameCnt);
        }
        return maxCnt >= m;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        m = sc.nextInt();

        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++){
                grid[i][j] = sc.nextInt();
            }
        }
        int happyCnt = 0;

        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++)
                seq[j] = grid[i][j];
            if(isHappy())
                happyCnt++;
        }

        for(int j=0; j<n; j++){
            for(int i=0; i<n; i++)
                seq[i] = grid[i][j];
            if(isHappy())
                happyCnt++;
        }

        System.out.print(happyCnt);
    }
}